### DISCUSSION TITLE
Edge fix option for zoe anything n model

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#93] [General]
I find the "Zoe Any N model" to be very good. in many situations it looks like it has better 3D than any large. however, the one issue with Zoe is some of the warping artifacts, and I believe if it had an edge fix option like the any large model, it would be superior. If possible, can you please add an edge fix option under those zoe models. Thanks.

**Comment by nagadomi** [comment]
I will enable Edge Fix to be used with `ZoeD_*` models.
Also, there seems to be a bug in `ZoeD_*` models where the depth aspect ratio is not correct for vertical images(height > width), so I will  fix that as well.

**Reply by kaelsonofkrypto** [reply]
thanks so much

**Reply by nagadomi** [reply]
I've made this change on the dev branch. Also fixed a bug in depth resizing.

I think theoretically ZoeD_Any_N is the best model. In my understanding, metric depth models do not underestimate slight differences in depth at the far end.
However, ZoeD_Any_N has an overshooting problem(like unsharp mask effect) at the foreground/background boundary.
I guess this is due to the use of blur data augmentation for training. (unsharp mask is inverse effect of blur)
I am not sure if the current Edge Fix (edge dilation) is able to reduce this type of artifact.

**Reply by kaelsonofkrypto** [reply]
![Error](https://github.com/nagadomi/nunif/assets/164638883/326e07ce-0316-469e-938c-6e708d11e46a)


Hey Nagadomi, after i tried to update im getting a Python int too large to convert to C long message the original file was 3840x1600. I then tried to see if lowering the resolution would work at 1080p and still didnt work. I even tried al the lowest settings in iw3 but i got that error message that says "Python int too large to convert to C long". I tried doing a fresh install and using the update. bat, but still can't figure it out, im using an rtx 3070 on this pc

**Reply by nagadomi** [reply]
No errors on my pc.
Could you try the following?
1. run `nunif-prompt.bat`
2. type `python -m iw3.gui` on the prompt and press ENTER
3. convert with the launched iw3-gui
4. if an error occurs, details will be displayed on the prompt

![prompt-run](https://github.com/nagadomi/nunif/assets/287255/f5b92854-fc11-4349-ba25-6f5a6c4aac80)


Also, the error occurs with all video or image? or only certain video?
I am guessing the error is related to video frame format conversion.

**Reply by kaelsonofkrypto** [reply]
![image](https://github.com/nagadomi/nunif/assets/164638883/b5b7a846-ba06-49d2-93a2-42c2fbdf3443)

hi this is a screenshot of the error message it gave me when i ran the prompt as you advised, i tried a few other files and still got an error

**Reply by nagadomi** [reply]
Fixed. Run `update.bat` again and try it again.
I added fraction representation for the following special FPS
```
29.97 = 30000 / 1001
23.976 = 24000 / 1001
59.94 = 60000 / 1001
```
FPS values that cannot be represented as fractions in the 32-bit integer range will result in that error.

**Reply by kaelsonofkrypto** [reply]
Thanks appreciate it, its working now after the last update you did,  your awesome

**Comment by loawizard** [comment]
yeah that would be great!

**Comment by nagadomi** [comment]
Sorry, the previous version of "Edge Fix for Zoedepth" was completely broken. 
Fixed in the latest code.

**Reply by kaelsonofkrypto** [reply]
Thank you so much for doing that I appreciate it
